synapse/lib/json.py

```diff
@@ -0,0 +1,231 @@
+
+    Similar to the standard library json.load().
+
+    Arguments:
+        fp (file): The python file pointer to read the data from.
+
+    Returns:
+        (object): The deserialized JSON data.
+
+    Raises:
+        synapse.exc.BadJsonText: This exception is raised when there is an error
+            deserializing the provided data.
+    '''
+    return loads(fp.read())
+
+def dumpsb(obj, sort_keys=False, indent=False, default=None, newline=False):
+    '''
+    Serialize a python object to byte string.
+
+    Similar to the standard library json.dumps().
+
+    Arguments:
+        obj (object): The python object to serialize.
+        sort_keys (Optional[bool]): Sort dictionary keys. Default: False.
+        indent (Optional[bool]): Include 2 spaces of indentation. Default: False.
+        default (Optional[callable]): Callback for serializing unknown object types. Default: None.
+        newline (Optional[bool]): Append a newline to the end of the serialized data. Default: False.
+
+    Returns:
+        (str | bytes): The JSON serialized python object.
+
+    Raises:
+        synapse.exc.MustBeJsonSafe: This exception is raised when a python object cannot be serialized.
+    '''
+    opts = 0
+
+    if indent:
+        opts |= orjson.OPT_INDENT_2
+
+    if sort_keys:
+        opts |= orjson.OPT_SORT_KEYS
+
+    if newline:
+        opts |= orjson.OPT_APPEND_NEWLINE
+
+    try:
+        return orjson.dumps(obj, option=opts, default=default)
+
+    except TypeError:
+        indent = 2 if indent else None
+
+        try:
+            ret = json.dumps(obj, sort_keys=sort_keys, indent=indent)
+        except TypeError as exc:
+            raise s_exc.MustBeJsonSafe(mesg=exc.args[0])
+
+        if newline:
+            ret += '\n'
+
+        return ret.encode()
+
+    except orjson.JSONEncodeError as exc:
+        raise s_exc.MustBeJsonSafe(mesg=exc.args[0])
+
+def dumps(obj, sort_keys=False, indent=False, default=None, asbytes=False, newline=False):
+    '''
+    Serialize a python object to a string.
+
+    Similar to the standard library json.dumps().
+
+    Arguments:
+        obj (object): The python object to serialize.
+        sort_keys (Optional[bool]): Sort dictionary keys. Default: False.
+        indent (Optional[bool]): Include 2 spaces of indentation. Default: False.
+        default (Optional[callable]): Callback for serializing unknown object types. Default: None.
+        newline (Optional[bool]): Append a newline to the end of the serialized data. Default: False.
+
+    Returns:
+        (str | bytes): The JSON serialized python object.
+
+    Raises:
+        synapse.exc.MustBeJsonSafe: This exception is raised when a python object cannot be serialized.
+    '''
+    data = dumpsb(obj, sort_keys=sort_keys, indent=indent, default=default, newline=newline)
+    return data.decode('utf8', errors='surrogatepass')
+
+def dumpb(obj, fp, sort_keys=False, indent=False, default=None, newline=False):
```

Comment:
Lets just make `dump()` support only binary and do away with the split use case. Any test instances can be converted to using a binary IO object :)